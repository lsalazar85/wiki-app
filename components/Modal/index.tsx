import { useEffect } from "react";
import { AnimatePresence } from "framer-motion";
import { ModalProps } from "../../interfaces";

import {
  Container,
  Positioned,
  Content,
  Header,
  CloseIcon,
  Scrollable,
  Background,
} from "./styles";

const Modal = ({ visible, title, onClose, children }: ModalProps) => {
  useEffect(() => {
    const [body] = document.getElementsByTagName("body");

    if (visible) {
      body.style.overflowY = "hidden";
    }

    return () => {
      body.style.overflowY = "auto";
    };
  }, [visible]);

  return (
    <AnimatePresence>
      {visible && (
        <Container>
          <Positioned>
            <Background
              onClick={() => onClose()}
              initial={{ opacity: 0 }}
              animate={{ opacity: 0.5 }}
              exit={{ opacity: 0 }}
            />
            <Content
              key="content"
              animate={{ y: 0 }}
              initial={{ y: "75%" }}
              exit={{ y: "100%" }}
              transition={{ type: "tween" }}
            >
              <Header>
                {title}
                <CloseIcon onClick={() => onClose()} />
              </Header>
              <Scrollable>{children}</Scrollable>
            </Content>
          </Positioned>
        </Container>
      )}
    </AnimatePresence>
  );
};

export default Modal;
