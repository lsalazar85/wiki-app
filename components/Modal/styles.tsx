import styled from 'styled-components';
import { motion } from 'framer-motion';
import { Times } from '@styled-icons/fa-solid';
import colors from '../../styles/colors';

export const Positioned = styled.div`
  position: relative;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
`;

export const Container = styled.div`
  position: fixed;
  left: 0;
  top: 0;
  width: 100vw;
  height: ${() => window.innerHeight + 'px'};
  z-index: 1000;
`;

export const Background = styled(motion.div)`
  position: absolute;
  width: 100vw;
  height: 100vh;
  background: ${colors.Black};
`;

export const Content = styled(motion.div)`
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;
  background: ${colors.White};
  padding: 1.87em 1em;
  border-radius: 0.5rem;
  @media (min-width: 1024px) {
    width: 25.625rem;
    left: calc(50% - 12.813rem);
    top: 50%;
    bottom: auto;
    transform: translateY(-50%) !important;
  }
`;

export const CloseIcon = styled(Times).attrs({
  size: '1.5em',
})`
  color: ${colors.Salmon};
  cursor: pointer;
`;

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  font-weight: 700;
  color: ${colors.Black};
  margin-bottom: 1.5rem;
`;

export const Scrollable = styled.div`
  max-height: 75vh;
  overflow-y: auto;
  ::-webkit-scrollbar {
    width: 0.2rem;
  }
  ::-webkit-scrollbar-thumb {
    background: ${colors.Salmon};
  }
`;
