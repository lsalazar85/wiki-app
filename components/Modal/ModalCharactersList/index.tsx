import { DetailsList, DetailText } from "../../Card/styles";
import { useArrayUrls } from "../../../hooks/useShowArrayUrls";

import { ModalListProps } from "../../../interfaces";

const ModalCharactersList = ({ id }: ModalListProps) => {
  const { data } = useArrayUrls("character-episode", `${id}`, "character");

  return (
    <DetailsList>
      <DetailText>{data?.name}</DetailText>
    </DetailsList>
  );
};

export default ModalCharactersList;
