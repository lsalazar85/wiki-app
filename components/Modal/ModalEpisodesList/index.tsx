import { DetailsList, DetailText } from "../../Card/styles";
import { ModalListProps } from "../../../interfaces";
import { useArrayUrls } from "../../../hooks/useShowArrayUrls";

const ModalEpisodesList = ({ id }: ModalListProps) => {
  const { data } = useArrayUrls('episode-characters',`${id}`,'episode');

  return (
    <DetailsList>
      <DetailText>Episode Nro: {data?.episode}</DetailText>
      <DetailText>Episode Name: {data?.name}</DetailText>
      <DetailText>Air Date: {data?.air_date}</DetailText>
    </DetailsList>
  );
};

export default ModalEpisodesList;
