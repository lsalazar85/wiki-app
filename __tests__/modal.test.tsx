/**
 * @jest-environment jsdom
 */

import { render  } from '@testing-library/react'
import Modal from "../components/Modal";


describe('Modal', () => {
    it('should render successfully', () => {
        const { asFragment } = render(
            <Modal
                onClose={() => jest.fn()}
                title='Modal Test'
                visible={true}
            >
                <div>Content</div>
            </Modal>
        )

        expect(asFragment()).toBeDefined()
    })
})
