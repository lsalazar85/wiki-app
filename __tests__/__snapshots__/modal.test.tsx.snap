// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render successfully 1`] = `
<DocumentFragment>
  <div
    class="styles__Container-sc-1gw39cm-1 bbYbUE"
  >
    <div
      class="styles__Positioned-sc-1gw39cm-0 YyMDn"
    >
      <div
        class="styles__Background-sc-1gw39cm-2 hEMrXN"
        style="opacity: 0;"
      />
      <div
        class="styles__Content-sc-1gw39cm-3 gRbZeu"
        style="transform: translateY(75%) translateZ(0);"
      >
        <div
          class="styles__Header-sc-1gw39cm-5 gtTjli"
        >
          Modal Test
          <svg
            aria-hidden="true"
            class="StyledIconBase-ea9ulj-0 bsimgx styles__CloseIcon-sc-1gw39cm-4 gjtfyv"
            fill="currentColor"
            focusable="false"
            height="1.5em"
            viewBox="0 0 352 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m242.72 256 100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="styles__Scrollable-sc-1gw39cm-6 gdvgUq"
        >
          <div>
            Content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
